libmozdata/bugzilla.py

```diff
@@ -453,20 +455,46 @@ def __get_bugs_by_search(self):
         """
         url = Bugzilla.API_URL + '?'
         header = self.get_header()
+        specials = {'count_only', 'limit', 'order', 'offset'}
         for query in self.bugids:
             if isinstance(query, six.string_types):
                 url = Bugzilla.API_URL + '?' + query
-                params = None
-            else:
+                self.bugs_results.append(self.session.get(url,
+                                                          headers=header,
+                                                          verify=True,
+                                                          timeout=self.TIMEOUT,
+                                                          background_callback=self.__bugs_cb))
+            elif not (specials & set(query.keys())):
                 url = Bugzilla.API_URL
-                params = query
-
-            self.bugs_results.append(self.session.get(url,
-                                                      params=params,
-                                                      headers=header,
-                                                      verify=True,
-                                                      timeout=self.TIMEOUT,
-                                                      background_callback=self.__bugs_cb))
+                params = query.copy()
+                params['count_only'] = 1
+                r = requests.get(url,
+                                 params=params,
+                                 headers=header,
+                                 verify=True,
+                                 timeout=self.TIMEOUT)
+                if r.status_code == 200:
+                    count = r.json()['bug_count']
+                    del params['count_only']
```

Comment:
You could copy query here, so you don't have to del this and you can avoid copying it in the following loop

Reply:
I can't remove the copy in the loop since 'offset' is changed.

Reply:
I assumed session.get would not hold a ref to the object, but we can't be sure, so yes better to keep copying it.

Reply:
I already had some issues when changing a shared dict between multiple requests